repos/system_upgrade/el7toel8/actors/checkrhui/actor.py

```diff
@@ -0,0 +1,59 @@
+from leapp.actors import Actor
+from leapp.libraries.common.rpms import has_package
+from leapp.models import (
+    InstalledRPM,
+    RHUIInfo,
+    RequiredTargetUserspacePackages,
+    DNFPluginInfo
+)
+from leapp.reporting import Report, create_report
+from leapp import reporting
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp.libraries.common import rhsm, rhui
+
+
+class CheckRHUI(Actor):
+    """
+    Check if system is using RHUI infrastructure (on public cloud) and send messages to
+    provide additional data needed for upgrade.
+    """
+
+    name = 'checkrhui'
+    consumes = (InstalledRPM)
+    produces = (RHUIInfo, RequiredTargetUserspacePackages, Report, DNFPluginInfo)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        for k, v in rhui.RHUI_CLOUD_MAP.items():
```

Comment:
@Rezney I think it is better to hav the variables names more verbose

Reply:
@ZhukovGreen thanks for taking a look. I think "k", "v" can be an exception as they are well known. But if you insist :)

Anyway, I will take all you comments into consideration but please have in mind this is "WIP" and some parts can still change. I was, atm, mainly interested in the general opinion on this approach.

Reply:
yep, sure